dev/com.ibm.ws.transport.http/test/com/ibm/ws/http/channel/test/api/SameSiteCookieUtilsTest.java

```diff
@@ -0,0 +1,250 @@
+     * if they were marked SameSite=Strict. This has been fixed on newer versions of
+     * iOS and MacOS.
+     *
+     */
+    @Test
+    public void testMacOSAndIOSCompatibility() {
+        try {
+
+            String userAgent;
+
+            //Test Safari @ MacOS 10.14 - Expected to return true.
+            userAgent = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/12.0 Safari/605.1.15";
+            assertTrue(SameSiteCookieUtils.isSameSiteNoneIncompatible(userAgent));
+
+            //Test Embedded @ MacOS 10.14 - Expected to return true
+            userAgent = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14) AppleWebKit/537.36 (KHTML, like Gecko)";
+            assertTrue(SameSiteCookieUtils.isSameSiteNoneIncompatible(userAgent));
+
+            //Test Safari  @ MacOS after 10.14 - Expected to return false
+            userAgent = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15) AppleWebKit/601.1.39 (KHTML, like Gecko) Version/10.1.2 Safari/601.1.39";
+            assertFalse(SameSiteCookieUtils.isSameSiteNoneIncompatible(userAgent));
+
+            //Test Safari @ MacOS 10.15.1 - major.minor.release version format on a supported user-agent
+            userAgent = "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_1) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.0.3 Safari/605.1.15";
+            assertFalse(SameSiteCookieUtils.isSameSiteNoneIncompatible(userAgent));
+
+            //Test Multiple Browsers in iOS 12 - Expected to return true for any browser.
+            // Safari
+            userAgent = "Mozilla/5.0 (iPhone; CPU iPhone OS 12_0 like Mac OS X) AppleWebKit/ 604.1.21 (KHTML, like Gecko) Version/ 12.0 Mobile/17A6278a Safari/602.1.26";
+            assertTrue(SameSiteCookieUtils.isSameSiteNoneIncompatible(userAgent));
+            //Chrome
+            userAgent = "Mozilla/5.0 (iPhone; CPU iPhone OS 12_0 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) CriOS/70.0.3538.75 Mobile/15E148 Safari/605.1";
+            assertTrue(SameSiteCookieUtils.isSameSiteNoneIncompatible(userAgent));
+            //Firefox
+            userAgent = "Mozilla/5.0 (iPad; CPU OS 12_0 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) FxiOS/13.2b11866 Mobile/16A366 Safari/605.1.15";
+            assertTrue(SameSiteCookieUtils.isSameSiteNoneIncompatible(userAgent));
+
+            //Test a browser after iOS version 12 (@ IOS 13)
+            userAgent = "Mozilla/5.0 (iPhone; CPU iPhone OS 13_0 like Mac OS X) AppleWebKit/602.1.38 (KHTML, like Gecko) Version/66.6 Mobile/14A5297c Safari/602.1";
+            assertFalse(SameSiteCookieUtils.isSameSiteNoneIncompatible(userAgent));
+
+            //TODO: considerations
+            //Test user agent on Safari and macOS before 10.14
+
+            //Test user agent on MacEmbedded browser and macOS before 10.14
+            //Test user agent on MacEmbedded browser and macOS after 10.14
+            //Test user agent on iOS before 12
```

Comment:
If we're considering these browsers as incompatible it would be good to add the tests if possible